src/main.rs

```diff
@@ -3,20 +3,75 @@
 //! Running this code starts a HTTP server instance. The default port is 3000, but you can set your
 //! own favorite port by exporting the `PORT` environment variable.
 
+extern crate rustc_serialize;
+extern crate docopt;
+
 extern crate spaceapi;
 extern crate spaceapi_server;
 
 use std::sync::{Mutex,Arc};
-
 use std::net::Ipv4Addr;
+use std::str::FromStr;
+
+use docopt::Docopt;
+
 use spaceapi::{Status, Location, Contact, Optional};
 use spaceapi_server::SpaceapiServer;
 use spaceapi_server::datastore::DataStore;
 use spaceapi_server::redis_store::RedisStore;
 
 
+static USAGE: &'static str = "
+Usage: coredump-status [-p PORT] [-i IP]
+
+Options:
+    -p PORT  The port to listen on [default: 3000].
+    -i IP    The ipv4 address to listen on [default: 127.0.0.1].
+";
+
+// We have to create a wrapper because we cannot implement Decodable for types
+// outside this crate (E0117).
+#[derive(Debug)]
+struct Ipv4 {
+    ip: Ipv4Addr,
+}
```

Comment:
Hmm this is a pitty, but isn't there an easier solution? See also https://github.com/rust-lang/rfcs/issues/493 for some more information.